src/wallet/db.cpp

```diff
@@ -531,11 +529,10 @@ bool BerkeleyBatch::Rewrite(BerkeleyDatabase& database, const char* pszSkip)
     while (true) {
         {
             LOCK(cs_db);
-            if (!env->mapFileUseCount.count(strFile) || env->mapFileUseCount[strFile] == 0) {
+            if (database.m_refcount == 0) {
                 // Flush log data to the dat file
                 env->CloseDb(strFile);
                 env->CheckpointLSN(strFile);
-                env->mapFileUseCount.erase(strFile);
```

Comment:
In commit "walletdb: track database file use as m_refcount within BerkeleyDatabase" (9cfb631f808aa6ccb10ab31d4b8325bd940a8916)

Note: seems this commit drops distinction between use count being 0 and use count not existing (no map entry). I think 0 use count meant the database wasn't in use but wasn't flushed yet, and no use count meant it was flushed. I think the last piece of code which used this distinction was the log_archive code removed in the previous commit 1dbc24d27d4b54b45c6b49ba41b4f64d0093dd71